src/commands/language.ts

```diff
@@ -0,0 +1,89 @@
+import { botCache } from "../../deps.ts";
+import { PermissionLevels } from "../types/commands.ts";
+import { createSubcommand, createCommand, getCurrentLanguage } from "../utils/helpers.ts";
+import { Embed } from "../utils/Embed.ts";
+import { db } from "../database/database.ts";
+
+const allowedLanguages = [ { id: "en_US", flag: ":flag_us:", name: "English"}, { id: "cs_CZ", flag:":flag_cz:", name: "Czech"}];
+
+createCommand({
+  name: "language",
+  aliases: ['lang'],
+  arguments: [
+    {
+      name: "subcommand",
+      type: "subcommand",
+      required: false,
+    },
+  ],
+  guildOnly: true,
+  permissionLevels: [PermissionLevels.MEMBER],
+  execute: async function (message) {
+    //@ts-ignore
+    const currentLanguageId = getCurrentLanguage(message.guildID, message.guild.preferredLocale);
+    let currentLanguage = allowedLanguages.find(item => item.id === currentLanguageId) || allowedLanguages[0];
```

Comment:
This is strange.
1. why use let
2. Why `|| allowedLanguages[0];` does TS give error here?

Reply:
![image](https://user-images.githubusercontent.com/71568768/104854669-f5e0fe80-58ff-11eb-8334-e9725e4f5eeb.png)
`|| allowedLanguages[0]` is not needed. my bad.

Reply:
`getCurrentLanguage` should not return a undefined language and also it should return a language that fits `allowedLanguages`

Reply:
`getCurrentLanguage` cannot return undefined. I do not know what do about this: `also it should return a language that fits allowedLanguages`

Reply:
`getCurrentLanguage` should return a language that is also in the `allowedLanguages` Array nothing else

Reply:
Would it be fine like this ?
```js
export function getCurrentLanguage(guildID: string, preferredLocale: string | undefined) {
  return botCache.guildLanguages.get(guildID) || preferredLocale;
}
```

Reply:
You need to return the default language if none is set

Reply:
Just to clarify this is not the right solution ?
```ts
export function getCurrentLanguage(guildID: string, preferredLocale: string | undefined) {
  return botCache.guildLanguages.get(guildID) || preferredLocale || 'en_US';
}
```

Reply:
If you can set the default lang in some config than no.
Also I'd like to see this: `preferredLocale?: string`

Reply:
You cannot set it anywhere else rn.
Gonna do this `preferredLocale?: string`.

Reply:
Should I add something like `defaultLanguage` variable to `config.ts` so the default language can be set in bot configuration file? Right know it just defaults to english.

Reply:
I think Skillz did not want this

Reply:
In that case it will just default to `en_US` like this.  I do not know a better solution.
```js
return botCache.guildLanguages.get(guildID) || preferredLocale || 'en_US';
```

Reply:
LGTM. Only thing I am not sure about is if we should only require the guildID and get the preferredLocale from cache.

Reply:
It could be done like this I think. I am not sure which solution is better. I am gonna leave that up to you and others to decide.
```ts
export function getCurrentLanguage(guildID: string) {
  return botCache.guildLanguages.get(guildID) || cache.guilds.get(guildID)?.preferredLocale || 'en_US';
}
```

Reply:
Well I personally I like that more.

Reply:
Alright than, changed it to this.